lib/fluent/plugin/out_file.rb

```diff
@@ -59,7 +59,7 @@ def symlink_path=(path)
 
       def generate_chunk(metadata)
         chunk = super
-        latest_chunk = metadata_list.sort_by(&:timekey).last
+        latest_chunk = metadata_list.select{|m| m.timekey }.sort_by(&:timekey).last
```

Comment:
Add comment is better for understanding why we pick up only chunks which have timekey data.